src/ansys/tools/visualization_interface/backends/pyvista/picker.py

```diff
@@ -0,0 +1,201 @@
+# Copyright (C) 2024 - 2025 ANSYS, Inc. and/or its affiliates.
+# SPDX-License-Identifier: MIT
+#
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""Module for managing picking and hovering of objects in a PyVista plotter."""
+from typing import TYPE_CHECKING, Union
+
+from ansys.tools.visualization_interface.types.edge_plot import EdgePlot
+from ansys.tools.visualization_interface.types.mesh_object_plot import MeshObjectPlot
+from ansys.tools.visualization_interface.utils.color import Color
+
+if TYPE_CHECKING:
+    import numpy as np
+    from pyvista import Actor
+
+    from ansys.tools.visualization_interface.backends.pyvista.pyvista import Plotter
+
+class AbstractPicker:
+    """Abstract base class for pickers."""
+
+    def __init__(self):
+        """Initialize the AbstractPicker class."""
+        pass
+
+    def pick_select_object(self, custom_object: Union[MeshObjectPlot, EdgePlot], pt: "np.ndarray") -> None:
+        """Determine actions to take when an object is selected."""
+        pass
+
+    def pick_unselect_object(self, custom_object: Union[MeshObjectPlot, EdgePlot]) -> None:
+        """Determine actions to take when an object is unselected."""
+        pass
+
+    def hover_select_object(self, custom_object: Union[MeshObjectPlot, EdgePlot], pt: "np.ndarray") -> None:
+        """Determine actions to take when an object is hovered over."""
+        pass
+
+    def hover_unselect_object(self, custom_object: Union[MeshObjectPlot, EdgePlot]) -> None:
+        """Determine actions to take when an object is unhovered."""
+        pass
+
+    def picked_dict(self) -> dict:
+        """Return the dictionary of picked objects."""
+        pass
```

Comment:
I would use the ``@abstractmethod`` decorator to force the implementation in any child class